Unable to Build

Just cloned,

ran

pnpm install
pnpm build

get the following error:

> packages/schema build$ pnpm build:ts && node ./dist/scripts/compile-json-schema.js
│ > @powersync/service-schema@1.16.3 build:ts C:\Users\NickWills\source\repos\powersync-service\packages\schema
│ > tsc -b
│ node:internal/fs/promises:957
│   const result = await PromisePrototypeThen(
[1 lines collapsed]
│ Error: ENOENT: no such file or directory, scandir 'C:\C:\Users\NickWills\source\repos\powersync-service\packages\jpgwire\ca'
│     at async Module.readdir (node:internal/fs/promises:957:18)
│     at async loadDefaultCertificates (file:///C:/Users/NickWills/source/repos/powersync-service/packages/jpgwire/dist/certs.js:6:19)
│2    at async file:///C:/Users/NickWills/source/repos/powersync-service/packages/jpgwire/dist/certs.js:3:30 {
│ node:internal/fs/promises:957
│3  const result = await PromisePrototypeThen(
│   const result = await PromisePrototypeThen(
│4                 ^
│                  ^
│5Error: ENOENT: no such file or directory, scandir 'C:\C:\Users\NickWills\source\repos\powersync-service\packages\jpgwire\ca'
│ Error: ENOENT: no such file or directory, scandir 'C:\C:\Users\NickWills\source\repos\powersync-service\packages\jpgwire\ca'
│ > @powersync/service-schema@1.16.3 build:ts C:\Users\NickWills\source\repos\powersync-service\packages\schema
│ > tsc -b
│ node:internal/fs/promises:957
│   const result = await PromisePrototypeThen(
│                  ^
│ Error: ENOENT: no such file or directory, scandir 'C:\C:\Users\NickWills\source\repos\powersync-service\packages\jpgwire\ca'
│     at async Module.readdir (node:internal/fs/promises:957:18)rvice\\packages\\jpgwire\\ca'
│     at async loadDefaultCertificates (file:///C:/Users/NickWills/source/repos/powersync-service/packages/jpgwire/dist/certs.js:6:19)
│     at async file:///C:/Users/NickWills/source/repos/powersync-service/packages/jpgwire/dist/certs.js:3:30 {
│   errno: -4058,
│   code: 'ENOENT',
│   syscall: 'scandir',
│   path: 'C:\\C:\\Users\\NickWills\\source\\repos\\powersync-service\\packages\\jpgwire\\ca'
│ }
│ Node.js v22.19.0
└─ Failed in 3.7s at C:\Users\NickWills\source\repos\powersync-service\packages\schema
service build$ tsc -b
└─ Done in 484ms
C:\Users\NickWills\source\repos\powersync-service\packages\schema:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @powersync/service-schema@1.16.3 build: `pnpm build:ts && node ./dist/scripts/compile-json-schema.js`
Exit status 1
 ELIFECYCLE  Command failed with exit code

--- a/packages/jpgwire/src/certs.ts
+++ b/packages/jpgwire/src/certs.ts
@@ -1,15 +1,16 @@
 import * as fs from 'fs/promises';
-import { resolve } from 'path';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
 
 export const DEFAULT_CERTS = await loadDefaultCertificates();
 
 async function loadDefaultCertificates() {
-  const dir = new URL('../ca', import.meta.url).pathname;
+  const dir = path.join(path.dirname(fileURLToPath(import.meta.url)), '../ca');
   const files = await fs.readdir(dir);
   let sum = '';
   for (let file of files) {
     if (file.endsWith('.pem')) {
-      sum += (await fs.readFile(resolve(dir, file), 'utf-8')) + '\n';
+      sum += (await fs.readFile(path.resolve(dir, file), 'utf-8')) + '\n';
     }
   }
   return sum;

